### PR TITLE
add location_type option to location choice provider

### PR DESCRIPTION
**Pull request by sravfeyn**
@pr33thi 

FYI @emord I noticed that some of ICDS UCR reports like MPR and ASR reports have location filters for state, district, block etc, but all choices show in those filters, so it's difficult to filter to just a district. I think this is not a big problem for lower location level locations, but is a pain for higher location users. I am surprised no one asked for this, but it's as easy as this to fix the issue. We could update various report sources with this, if it's good.

### CODE REVIEW

File: corehq/apps/userreports/README.md [hunk 1]
```diff
@@ -1289,6 +1290,7 @@ Example assuming "village" is a location ID, which is converted to names using t
       "type": "location",
       "include_descendants": true,
       "show_full_path": true
```
Comment (emord) [inline corehq/apps/userreports/README.md]: This needs a `,` at end of the line

File: corehq/apps/userreports/reports/filters/choice_providers.py [hunk 2]
```diff
@@ -229,20 +229,29 @@ def __init__(self, report, filter_slug):
...
+            locations = locations.filter_path_by_user_input(
                 domain=self.domain,
                 user_input=query_text
-            ).accessible_to_user(self.domain, user)
+            )
+
+        if self.location_type:
+            try:
+                locations = locations.filter(location_type__code=self.location_type)
+            except SQLLocation.DoesNotExist:
```
Comment (emord) [inline corehq/apps/userreports/reports/filters/choice_providers.py]: When would this exception be hit? I believe the previous line should be returning a queryset, but I think `DoesNotExist` is only thrown when trying to access an instance of a model that doesn't exist, so I'm not sure what this exception will catch

**Review by emord — COMMENTED** [review]
I've noticed this as well, but wasn't sure why it was like this. I think its a fine change to make

File: corehq/apps/userreports/reports/filters/choice_providers.py (hunk 2 above)
Comment (sravfeyn) [reply corehq/apps/userreports/reports/filters/choice_providers.py]: Ha, true, I will remove this

**Review by pr33thi — APPROVED** [review]

**Review by emord — APPROVED** [review]
you may need to merge master to this branch to get tests to run

**Review by emord — APPROVED** [review]

**Comment by sravfeyn** [comment]
The build fails for random reasons or takes long time. I am merging master again.

**Pull request merged**